src/mlpack/core/optimizers/nestrov/nag.hpp

```diff
@@ -0,0 +1,155 @@
+/**
+ * @file nag.hpp
+ * @author Ryan Curtin
+ * @author Krishna Kant Singh
+ *
+ * Stochastic Gradient Descent(SGD) with Nestrov Accelrated Gradient
+ * update(NAG) is an approach 
+ * that enjoys accelerated convergence rates compared to vanilla SGD.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_MOMENTUM_SGD_MOMENTUM_SGD_HPP
```

Comment:
The header guard should be named like:  ``MLPACK_CORE_OPTIMIZERS_NESTROV_NAG_HPP``.